src/tablestore.rs

```diff
@@ -371,4 +401,30 @@ mod tests {
         )
         .await;
     }
+
+    #[tokio::test]
+    async fn test_wal_write_should_fail_when_fenced() {
+        let os = Arc::new(object_store::memory::InMemory::new());
+        let format = SsTableFormat::new(32, 1, None);
+        let ts = Arc::new(TableStore::new(os.clone(), format, Path::from(ROOT)));
+        let wal_id = SsTableId::Wal(1);
+
+        // write a wal sst
+        let mut sst1 = ts.table_builder();
+        sst1.add(b"key", Some(b"value")).unwrap();
+        let table = sst1.build().unwrap();
+        ts.write_sst(&wal_id, table).await.unwrap();
+
+        let mut sst2 = ts.table_builder();
+        sst2.add(b"key", Some(b"value")).unwrap();
+        let table2 = sst2.build().unwrap();
+
+        // write another walsst with the same id.
+        let result = ts.write_sst(&wal_id, table2).await;
+
+        match result {
```

Comment:
you should be able to write something like `assert!(matches!(result, Err(error::SlateDBError::Fenced)))`